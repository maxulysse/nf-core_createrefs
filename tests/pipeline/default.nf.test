nextflow_pipeline {

    name "Test pipeline"
    script "main.nf"
    tag "default"

    test("Run default test") {

        when {
            params {
                outdir                       = "$outputDir"
                validationSchemaIgnoreParams = 'genomes,baseDir,base-dir,outputDir,output-dir'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/multiqc/").exists() },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert snapshot(path("$outputDir/reference/cnvkit/").list()).match("cnvkit") }
            )
        }
    }
}
