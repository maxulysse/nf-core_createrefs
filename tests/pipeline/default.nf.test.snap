{
    "software_versions": {
        "content": [
            "{CNVKIT_BATCH={cnvkit=0.9.10}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.7, yaml=5.4.1}, Workflow={nf-core/createpanelrefs=1.0dev}}"
        ],
        "timestamp": "2024-01-17T10:30:57.668525"
    },
    "cnvkit": {
        "content": [
            "panel.cnn:md5,07dea67088da689ad04012552c606882",
            "test.paired_end.sorted.antitargetcoverage.cnn:md5,203caf8cef6935bb50b4138097955cb8",
            "test.paired_end.sorted.targetcoverage.cnn:md5,ff526714696aa49bdc1dc8d00d965266",
            "test2.paired_end.sorted.antitargetcoverage.cnn:md5,203caf8cef6935bb50b4138097955cb8",
            "test2.paired_end.sorted.targetcoverage.cnn:md5,6ae6b3fce7299eedca6133d911c38fe1"
        ],
        "timestamp": "2023-07-08T16:47:57+0000"
    }
}